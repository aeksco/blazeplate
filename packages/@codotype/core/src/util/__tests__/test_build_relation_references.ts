import { inflateSchemas, inflateSchema } from "../inflate";
import { SchemaInput, testState, Relation, Schema } from "../../";

const { userSchema, movieSchema } = testState;

// // // //

const testCases: [string, SchemaInput, SchemaInput[]][] = [
    ["basic schemas", userSchema, [userSchema, movieSchema]],
];

// // // //

describe("/lib/inflateSchema.js", () => {
    testCases.forEach(testCase => {
        test(testCase[0], () => {
            const expectedSchema: Schema = inflateSchema({
                schemaInput: userSchema,
                relations: [],
            });
            expect(expectedSchema).toMatchSnapshot();
        });
    });
});

// // // //

describe("/lib/inflateSchemas.js", () => {
    testCases.forEach(testCase => {
        test(testCase[0], () => {
            const expectedSchemas: Schema[] = inflateSchemas({
                schemaInputs: [userSchema, movieSchema],
                relations: [],
            });
            expect(expectedSchemas).toMatchSnapshot();
        });
    });
});
