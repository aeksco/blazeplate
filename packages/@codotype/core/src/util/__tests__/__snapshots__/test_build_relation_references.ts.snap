// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`/lib/inflateSchema.js basic schemas 1`] = `
Object {
  "attributes": Array [],
  "configuration": Object {},
  "id": "12345",
  "identifiers": Object {
    "plural": Object {
      "camel": "users",
      "kebab": "users",
      "label": "Users",
      "pascal": "Users",
      "snake": "users",
    },
    "singular": Object {
      "camel": "user",
      "kebab": "user",
      "label": "User",
      "pascal": "User",
      "snake": "user",
    },
  },
  "references": Array [],
  "relations": Array [],
}
`;

exports[`/lib/inflateSchemas.js basic schemas 1`] = `
Array [
  Object {
    "attributes": Array [],
    "configuration": Object {},
    "id": "12345",
    "identifiers": Object {
      "plural": Object {
        "camel": "users",
        "kebab": "users",
        "label": "Users",
        "pascal": "Users",
        "snake": "users",
      },
      "singular": Object {
        "camel": "user",
        "kebab": "user",
        "label": "User",
        "pascal": "User",
        "snake": "user",
      },
    },
    "references": Array [],
    "relations": Array [],
  },
  Object {
    "attributes": Array [
      Object {
        "addons": Object {
          "required": true,
        },
        "datatype": "STRING",
        "defaultValue": null,
        "id": "name-attr",
        "identifiers": Object {
          "camel": "name",
          "kebab": "name",
          "label": "Name",
          "pascal": "Name",
          "snake": "name",
        },
        "internalNote": "the name of the user",
        "locked": false,
        "source": "user",
      },
      Object {
        "addons": Object {
          "required": true,
        },
        "datatype": "STRING",
        "defaultValue": null,
        "id": "email-attr",
        "identifiers": Object {
          "camel": "email",
          "kebab": "email",
          "label": "Email",
          "pascal": "Email",
          "snake": "email",
        },
        "internalNote": "the email of the user",
        "locked": false,
        "source": "user",
      },
    ],
    "configuration": Object {},
    "id": "45678",
    "identifiers": Object {
      "plural": Object {
        "camel": "movies",
        "kebab": "movies",
        "label": "Movies",
        "pascal": "Movies",
        "snake": "movies",
      },
      "singular": Object {
        "camel": "movie",
        "kebab": "movie",
        "label": "Movie",
        "pascal": "Movie",
        "snake": "movie",
      },
    },
    "references": Array [],
    "relations": Array [],
  },
]
`;
